tests/tensor/rewriting/test_basic.py

```diff
@@ -1298,41 +1298,57 @@ def test_local_join_make_vector():
 
 
 def test_local_sum_make_vector():
+    # To check that rewrite is applied, we must enforce dtype to
+    # allow rewrite to occur even if floatX != "float64"
     a, b, c = scalars("abc")
+    point = {a: 1.0, b: 2.0, c: 3.0}
     mv = MakeVector(config.floatX)
-    output = mv(a, b, c).sum()
-
-    output = rewrite_graph(output)
-    between = vars_between([a, b, c], [output])
-    for var in between:
-        assert (var.owner is None) or (not isinstance(var.owner.op, MakeVector))
-
-    # Check for empty sum
+    output = mv(a, b, c).sum(dtype="float64")
+    rewrite_output = rewrite_graph(output)
+    # check rewrite
+    assert not equal_computations([output], [rewrite_output])
```

Comment:
The idea with equal_computations is to write the expected graph explicitly and assert it matches with the rewritten, nor just that it has changed.

Reply:
I'm not sure that I understand. This test is simply checking that a rewrite has been applied, such that the computation of `output` is different than the computation of `rewrite_output`. What are you suggesting?

Reply:
Only checking it's different is very vague. It could be different in many ways that are not correct (for instance just changed the order of the vector being summed). I'm suggesting testing it has changed to the specific form we expect from the rewrite.

For example, instead of:

```python
x,y,z = pt.dscalars("x", "y", "z")
original_out = pt.sum([x, y, z], dtype="float32")
rewritten_out = ...
assert not equal_computations([original_out], [expected_out])
```

Do:
```python
x,y,z = pt.dscalars("x", "y", "z")
original_out = pt.sum([x, y, z], dtype="float32")
rewritten_out = ...
expected_out = cast(add(x, y, z), dtype="float32”)
assert equal_computations([rewritten_out], [expected_out])
```

Reply:
OK thanks for the suggestion!